### PR TITLE
add lax.associative_scan to docs

### PR DESCRIPTION
**Pull request by mattjj**
Also tweaked the docstring to say "arrays" instead of "Tensors".

### CODE REVIEW

File: jax/lax/lax_control_flow.py [hunk 1]
```diff
@@ -2120,33 +2120,28 @@ def associative_scan(fn, elems):
...
+      recursively applying ``fn`` to combine the first ``k`` elements of
+      ``elems``. For example, given ``elems = [a, b, c, ...]``, the result
+      would be ``[a, fn(a, b), fn(fn(a, b), c), ...]``.
 
-  np.associative_scan(operator.add, np.arange(0, 4))
-  # ==> [ 0, 1, 3, 6]
+  For example:
 
-  # Example 2: Partial products of random matrices.
+    np.associative_scan(operator.add, np.arange(0, 4))
```
Comment (fehiepsi) [inline jax/lax/lax_control_flow.py:2141]: nit: np -> lax

File: jax/lax/lax_control_flow.py [hunk 2]
```diff
@@ -2120,33 +2120,28 @@ def associative_scan(fn, elems):
...
+    # ==> [ 0, 1, 3, 6]
 
-  np.associative_scan(np.matmul, matrices)
-  ```
+    np.associative_scan(np.matmul, matrices)  # partial products of matrices
```
Comment (fehiepsi) [inline jax/lax/lax_control_flow.py:2144]: ditto: np -> lax

**Review by fehiepsi — COMMENTED** [review]
Oh, I didn't know that JAX has this feature. We used parallel scan a lot in time series predictions (e.g. in the implementation of Gaussian HMM in Pyro). This would be very helpful when we port Pyro implementations to NumPyro. Kudos to JAX team!

Just curious: IIUC, this does 2 steps up-sweep (up-pass) and down-sweep (down-pass). Does JAX have a utility to only do up-sweep part,  e.g. to compute the sum of an array?

**Comment by hawkinsp** [comment]
The docs appear to have a several rendering problems: https://jax--3583.org.readthedocs.build/en/3583/_autosummary/jax.lax.associative_scan.html#jax.lax.associative_scan

Please fix?

**Comment by mattjj** [comment]
Thanks both for the fixes! Also thanks @hawkinsp for showing me how to check docs rendering on a branch.

**Comment by jakevdp** [comment]
Closing in favor of #4172

**Pull request closed without merging**